File: backend-java/src/main/java/com/example/pricing/entity/TrafficPromoDaily.java
/*
 * 流量促销日数据实体，对应外部导入的投流和促销指标。
 */

package com.example.pricing.entity;

import com.baomidou.mybatisplus.annotation.FieldFill;
import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableField;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import lombok.Data;

import java.math.BigDecimal;
import java.time.LocalDate;
import java.time.LocalDateTime;

/**
 * 流量推广日报实体，记录投放渠道、曝光点击和投入产出表现。
 */
@Data
@TableName("traffic_promo_daily")
public class TrafficPromoDaily {
    @TableId(type = IdType.AUTO)
    private Long id;

    @TableField("shop_id")
    private Long shopId;

    @TableField("product_id")
    private Long productId;

    @TableField("stat_date")
    private LocalDate statDate;

    @TableField("traffic_source")
    private String trafficSource;

    @TableField("impression_count")
    private Integer impressionCount;

    @TableField("click_count")
    private Integer clickCount;

    @TableField("visitor_count")
    private Integer visitorCount;

    @TableField("cost_amount")
    private BigDecimal costAmount;

    @TableField("pay_amount")
    private BigDecimal payAmount;

    @TableField("roi")
    private BigDecimal roi;

    @TableField("upload_batch_id")
    private Long uploadBatchId;

    @TableField(value = "created_at", fill = FieldFill.INSERT)
    private LocalDateTime createdAt;
}

File: backend-python/app/models/traffic_promo_daily.py
"""流量与促销日指标模型，对应外部导入的流量促销数据。"""

from datetime import date, datetime
from decimal import Decimal

from sqlalchemy import BigInteger, Date, DateTime, Integer, String, text
from sqlalchemy.dialects.mysql import DECIMAL
from sqlalchemy.orm import Mapped, mapped_column

from app.db.base import Base


class TrafficPromoDaily(Base):
    __tablename__ = "traffic_promo_daily"

    id: Mapped[int] = mapped_column(BigInteger, primary_key=True, autoincrement=True)
    shop_id: Mapped[int] = mapped_column(BigInteger, nullable=False)
    product_id: Mapped[int | None] = mapped_column(BigInteger, nullable=True)
    stat_date: Mapped[date] = mapped_column(Date, nullable=False)
    traffic_source: Mapped[str] = mapped_column(String(50), nullable=False)
    impression_count: Mapped[int] = mapped_column(Integer, nullable=False, default=0, server_default="0")
    click_count: Mapped[int] = mapped_column(Integer, nullable=False, default=0, server_default="0")
    visitor_count: Mapped[int] = mapped_column(Integer, nullable=False, default=0, server_default="0")
    cost_amount: Mapped[Decimal] = mapped_column(DECIMAL(12, 2), nullable=False, default=0)
    pay_amount: Mapped[Decimal] = mapped_column(DECIMAL(12, 2), nullable=False, default=0)
    roi: Mapped[Decimal | None] = mapped_column(DECIMAL(10, 4), nullable=True, default=0)
    upload_batch_id: Mapped[int | None] = mapped_column(BigInteger, nullable=True)
    created_at: Mapped[datetime] = mapped_column(DateTime, nullable=False, server_default=text("CURRENT_TIMESTAMP"))
